### PR TITLE
[vividus] Add while-like step

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/commons/nav.adoc [hunk 1]
```diff
@@ -1,3 +1,4 @@
-.Tests development
+.Commons
+* xref:vividus-steps.adoc[Steps]
 * xref:tests-configuration.adoc[Tests Configuration]
```
Comment (valfirst) [inline docs/modules/commons/nav.adoc:3]: lets move it outside commons

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
+= Steps available out of the box
+
+Here one could find description of the steps that delivered with Vividus itself without any plugins required.
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:3]: `that are delivered`

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+=== While-like steps executor
+
+==== *_Info_*
+
+Executes steps while condition or iteration limit reached.
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:9]: `while` -> `until`

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+[IMPORTANT]
+If iteration limit reached no failure or exception will happen
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:12]: `is reached`

Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:12]: `happen` -> `occur`

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+==== *_Wording_*
+
+[source,gherkin]
+----
+When I execute steps at most $max times while `$variableName` is $comparisonRule `$expectedValue`:$stepsToExecute
+----
+
+==== *_Parameters_*
+
+. `$max` - iterations
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:23]: please elaborate

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+Executes steps while condition or iteration limit reached.
+
+[IMPORTANT]
+If iteration limit reached no failure or exception will happen
+
+==== *_Wording_*
+
+[source,gherkin]
+----
+When I execute steps at most $max times while `$variableName` is $comparisonRule `$expectedValue`:$stepsToExecute
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:18]:
```
... while variable '$name' is $comparisonRule...
```
lets make variable reference more explicit

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+----
+
+==== *_Parameters_*
+
+. `$max` - iterations
+. `$variableName` - variable name to check
```
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:24]: a name of the variable to check

File: docs/modules/plugins/pages/plugin-web-app.adoc [hunk 8]
```diff
@@ -11,7 +11,6 @@ The plugin provides functionality to interact with Web applications.
 implementation(group: 'org.vividus', name: 'vividus-plugin-web-app', version: '{page-component-version}')
 ----
 
-
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-web-app.adoc:14]: noisy change

File: docs/modules/commons/pages/vividus-steps.adoc (hunk 6 above)
Comment (valfirst) [inline docs/modules/commons/pages/vividus-steps.adoc:18]: I think we need to document alias as well

File: vividus/src/main/java/org/vividus/bdd/steps/ExecutableSteps.java [hunk 9]
```diff
@@ -81,6 +81,54 @@ public void ifVariableNotSetPerformSteps(String name, SubSteps stepsToExecute)
...
+        Matcher<String> valueVerifier = comparisonRule.getComparisonRule(expectedValue);
+        int iterationsLeft = max;
+        while (iterationsLeft > 0 && variableValueFits(name, valueVerifier))
+        {
+            --iterationsLeft;
+            stepsToExecute.execute(Optional.empty());
+        }
+    }
+
+    private boolean variableValueFits(String name, Matcher<String> verifier)
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/steps/ExecutableSteps.java:122]: `doesVariableValueMatch`

File: vividus/src/main/java/org/vividus/bdd/steps/ExecutableSteps.java [hunk 10]
```diff
@@ -81,6 +81,54 @@ public void ifVariableNotSetPerformSteps(String name, SubSteps stepsToExecute)
...
+    {
+        String variable = bddVariableContext.getVariable(name);
+        if (variable == null)
+        {
+            return true;
+        }
+        return verifier.matches(variable);
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/bdd/steps/ExecutableSteps.java:129]:
```java
return variable == null || verifier.matches(variable);
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/commons/pages/vividus-steps.adoc (hunk 6 above)
Comment (valfirst) [reply docs/modules/commons/pages/vividus-steps.adoc:22]: @ikalinin1 ☝️

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**